pulp_ansible/app/tasks/collectionversion_index.py

```diff
@@ -0,0 +1,179 @@
+    cvs = repository_version.content.filter(pulp_type="ansible.collection_version").values_list(
+        "pk", flat=True
+    )
+    cvs = CollectionVersion.objects.filter(pk__in=cvs)
+
+    # get the set of signatures in this repo version
+    repo_signatures_pks = repository_version.content.filter(
+        pulp_type="ansible.collection_signature"
+    ).values_list("pk", flat=True)
+    repo_signatures = CollectionVersionSignature.objects.filter(pk__in=repo_signatures_pks)
+
+    # get the set of deprecations in this repo version
+    deprecations = repository_version.content.filter(
+        pulp_type="ansible.collection_deprecation"
+    ).values_list("pk", flat=True)
+    deprecations = AnsibleCollectionDeprecated.objects.filter(pk__in=deprecations)
+    deprecations_set = {(x.namespace, x.name) for x in deprecations}
+
+    # find all namespaces in the repo version
+    namespaces = repository_version.content.filter(pulp_type="ansible.namespace").values_list(
+        "pk", flat=True
+    )
+    namespaces = AnsibleNamespace.objects.filter(pk__in=namespaces)
+    namespaces = {x.name: x for x in namespaces}
+
+    # map out the namespace(s).name(s) for everything in the repo version
+    colset = set(cvs.values_list("namespace", "name").distinct())
+
+    # clean up any repo+namespace+name indexes that no longer live in the repo ...
+    if not use_repository_version:
+        # CVIndex.objects.filter(repository=repository, repository_version=None).exclude(
+        #    collection_version__pk__in=cvs.values_list("pk", flat=True)
+        # ).delete()
+        pass
+
+    # iterate through each collection in the repository
+    for colkey in colset:
+        namespace, name = colkey
+
+        # get all the versions for this collection
+        related_cvs = cvs.filter(namespace=namespace, name=name).only(
+            "namespace", "name", "version"
+        )
+
+        # what is the "highest" version in this list?
+        highest_version = get_highest_version_string_from_cv_objects(related_cvs)
+
+        # should all of these CVs be deprecated?
+        is_deprecated = colkey in deprecations_set
+
+        # process each related CV
+        for rcv in related_cvs:
+            # get the related signatures for this CV
+            rcv_signatures = repo_signatures.filter(signed_collection=rcv).count()
+
+            repo_v = None
+            if use_repository_version:
+                repo_v = repository_version
+
+            # create|update the index for this CV
+            CVIndex.objects.update_or_create(
+                repository=repository,
+                repository_version=repo_v,
+                collection_version=rcv,
+                defaults={
+                    "is_highest": rcv.version == highest_version,
+                    "is_signed": rcv_signatures > 0,
+                    "is_deprecated": is_deprecated,
+                    "namespace_metadata": namespaces.get(rcv.namespace, None),
```

Comment:
```suggestion
                    "namespace_metadata": namespaces.get(namespace, None),
```
All of the related_cvs will have the same namespace.